scripts/check_firestore_symbols.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# DESCRIPTION: This script identifies Objective-C symbols within the
+# `FirebaseFirestore.xcframework` that are not automatically linked when used
+# in a client target. Because the `FirebaseFirestore.xcframework` should
+# function without clients needing to pass the `-ObjC`, this script catches
+# potential regressions that break that requirement.
+#
+# DEPENDENCIES: This script depends on the given Firebase repo's `Package.swift`
+# using the `FIREBASECI_USE_LOCAL_FIRESTORE_ZIP` env var to swap the Firestore
+# target definition out to instead reference a *local* binary using the
+# `.binaryTarget(path:)` API.
+#
+# DESIGN: This script creates an executable package that depends on Firestore
+# via a local binary SPM target. The package is built twice, once with the
+# -ObjC flag and once without. The linked Objective-C symbols are then
+# stripped from each build's resulting executable. The symbols are then diffed
+# to determine if there exists symbols that were only linked due to the -ObjC
+# flag.
+#
+# USAGE: ./check_firestore_symbols.sh <PATH_TO_FIREBASE_REPO> <PATH_TO_FIRESTORE_XCFRAMEWORK>
+
+if [ $# -ne 2 ]; then
```

Comment:
nit: `[[ $# -ne 2 ]]`

> `[[ ... ]]` is preferred over `[`, `test` and `/usr/bin/[`.

-- [Google Shell Style Guide](https://google.github.io/styleguide/shellguide.html#test----and---)